### PR TITLE
refactor(test, Android): add manual scenario for stack transition animations

### PR DESCRIPTION
**Pull request by LKuchno**
## Description
Closes: https://github.com/software-mansion/react-native-screens-labs/issues/1604

Adds a manual test **scenario** for the gamma/v5 `StackContainer` transition animations on Android, and renames the test directory to match the repo's naming convention.

The stack animation behavior is driven by the Android **Transition API** ([PR #3629](https://github.com/software-mansion/react-native-screens/pull/3629)). This scenario documents a manual QA pass that validates push, on-screen **Pop** and the interactive **predictive back gesture** all animate smoothly, and guards against the four regressions #3629 fixed (disappearing shadows, mid-transition content jumps, non-continuous animation, and native-pop completing instantly). It also checks that a **nested** `StackContainer` animates identically to the top-level one.

This scenario is **manual only** - animation smoothness, shadow continuity, absence of content jumps, and the interactive predictive back gesture are visual/interactive qualities Detox cannot assert reliably, so `e2eCoverage` is set to `incomplete`.

## Changelog

- Add `scenario.md` describing the manual QA flow for Android stack transition animations (baseline, push, on-screen pop, predictive back gesture, nested stack, and re-pushing the same route edge case).
- Rename `single-feature-tests/stack-v5/test-animation-android/` → `test-stack-animation-android/` and update the corresponding imports/exports in `stack-v5/index.ts`.
- Set `e2eCoverage` from `tbd` to `incomplete` in the scenario description, since this test cannot be automated with Detox (it is a strongly visual check).

### CODE REVIEW

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+## E2E test
+
+Incomplete: Not automated. Animation smoothness, shadow continuity, absence of content
+jumps, and the interactive predictive back gesture are visual / interactive
+qualities that Detox cannot assert reliably. This scenario is manual only.
+
+## Prerequisites
+
+- Android emulator or device (this is an **Android-only** test; there is no
+  iOS-specific behavior to verify here).
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md]:
```suggestion
- Android emulator or device
```
nit: not sure if we need this warning

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+- [ ] **Home** is shown with a solid **yellow** background and **Push Blue** /
+      **Push Red** / **Push NestedHost** buttons. No **Pop** button is shown.
+
+### Push animation
+
+2. Tap **Push Blue**.
+
+- [ ] The **Blue** screen slides from right to left 
+      in over **Home** with a smooth, continuous
+      animation. A drop shadow is visible along the leading edge of the
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md]: I don't think there is any shadow behind the screen. IIRC, the original issue was about the shadows inside screen's content (e.g. around the buttons).

Image

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+- [ ] The pop **completes smoothly from the current gesture position** to
+      fully showing **Blue** — the animation continues to completion rather
+      than snapping instantly.
+
+8. Push back to **Red**, start the same edge swipe, but **release before** the
+   commit threshold (a short drag).
+
+- [ ] The gesture is **cancelled smoothly**: **Red** animates back into place
+      and no navigation change occurs.
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md]: Maybe it's a skill issue but even if I drag a little bit, the pop finishes. I need to swipe back towards the edge so that the pop is canceled.

https://github.com/user-attachments/assets/6005de7b-b6a6-44f8-b636-c3d016ab2e7a

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md (hunk 2 above)
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md]: You are right!
Steps updated to check on button shadows 
https://github.com/software-mansion/react-native-screens/pull/4288/changes/be1c80a81368d3e32a6b6acbc24d00907c3f64b8

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md (hunk 3 above)
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md]: Right! It's not always coming back to red, i change the step to swipe back to te left edge 
https://github.com/software-mansion/react-native-screens/pull/4288/changes/be1c80a81368d3e32a6b6acbc24d00907c3f64b8

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+- [ ] The **Red** screen slides in over **Blue**, same as step 2 — smooth,
+      button shadows present, no content jump.
+
+### Pop animation (on-screen button)
+
+4. On **Red**, tap **Pop**.
+
+- [ ] **Red** slides back out and **Blue** is revealed with a smooth
+      animation (the reverse of the push). The button shadows stays visible;
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md]:
```suggestion
      animation (the reverse of the push). The button shadows stay visible;
```
nit

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+- [ ] The **Red** screen slides in over **Blue**, same as step 2 — smooth,
+      button shadows present, no content jump.
+
+6. On **Red** slowly swipe inward from the **left screen edge** and **hold** without
+   releasing.
+
+- [ ] The gesture is **interactive and continuous**: **Red** follows your
+      finger and the screen underneath (**Blue**) is progressively revealed,
+      tracking the drag position. The animation is not stepped or frozen, and
+      the button shadows remains visible.
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md]:
```suggestion
      the button shadows remain visible.
```
nit

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+- [ ] The nested host slides in and shows **NestedHome** (yellow) with **Push
+      NestedBlue** / **Push NestedRed** / **Pop** buttons. The push is
+      animated with the same quality as the outer stack.
+
+10. On **NestedHome**, tap **Push NestedBlue**, then on **NestedBlue** tap
+    **Push NestedRed**.
+
+- [ ] Each push inside the **nested** stack slides in smoothly, with a visible
+      button shadows and no content jump — identical behaviour to the top-level stack.
```
Comment (kligarski) [inline apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md]:
```suggestion
- [ ] Each push inside the **nested** stack slides in smoothly, with visible
      button shadows and no content jump — identical behaviour to the top-level stack.
```
nit

**Review by kligarski — APPROVED** [review]
Looks good, thank you!

File: apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md (hunk 6 above)
Comment (LKuchno) [reply apps/src/tests/single-feature-tests/stack-v5/test-stack-animation-android/scenario.md]: https://github.com/software-mansion/react-native-screens/pull/4288/changes/c00567f2ee00fc884ebdba60756cc6ee736d643a :)
